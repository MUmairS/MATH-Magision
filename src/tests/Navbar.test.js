import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { BrowserRouter as Router } from 'react-router-dom';
import Navbar from '../components/Navbar';

describe('renders correctly', () => {
  test('render Navbar', () => {
    const nav = render(
      <Router>
        <Navbar />
      </Router>,
    );
    expect(nav).toMatchSnapshot();
  });

  test('check title exist', () => {
    render(
      <Router>
        <Navbar />
      </Router>,
    );
    expect(
      screen.getByRole('heading', { name: /Math Magicians/i }),
    ).toBeInTheDocument();
  });
});
