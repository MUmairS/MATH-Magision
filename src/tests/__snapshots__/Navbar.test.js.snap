// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly render Navbar 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="navBar"
      >
        <h1
          class="title"
        >
          Math Magicians
        </h1>
        <ul
          class="linksContainer"
        >
          <li
            class="linkItems"
          >
            <a
              class="links"
              href="/"
            >
              Home
            </a>
          </li>
          <div
            class="separateDiv"
          />
          <li
            class="linkItems"
          >
            <a
              class="links"
              href="/calculator"
            >
              Calculator
            </a>
          </li>
          <div
            class="separateDiv"
          />
          <li
            class="linkItems"
          >
            <a
              class="links"
              href="/quote"
            >
              Quote
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="navBar"
    >
      <h1
        class="title"
      >
        Math Magicians
      </h1>
      <ul
        class="linksContainer"
      >
        <li
          class="linkItems"
        >
          <a
            class="links"
            href="/"
          >
            Home
          </a>
        </li>
        <div
          class="separateDiv"
        />
        <li
          class="linkItems"
        >
          <a
            class="links"
            href="/calculator"
          >
            Calculator
          </a>
        </li>
        <div
          class="separateDiv"
        />
        <li
          class="linkItems"
        >
          <a
            class="links"
            href="/quote"
          >
            Quote
          </a>
        </li>
      </ul>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
